lib/std/zig/system/NativePaths.zig

```diff
@@ -135,6 +135,48 @@ pub fn detect(allocator: Allocator, native_info: NativeTargetInfo) !NativePaths
         // zlib.h is in /usr/include (added above)
         // libz.so.1 is in /lib/x86_64-linux-gnu (added here)
         try self.addLibDirFmt("/lib/{s}", .{triple});
+
+        // NOTE: distro like guix doesn't use FHS, so it relies on envorinment
+        // variables (C_INCLUDE_PATH, CPLUS_INCLUDE_PATH and LIBRARY_PATH) to
+        // search for headers and libraries
+        if (process.getEnvVarOwned(allocator, "C_INCLUDE_PATH")) |c_include_path| {
+            defer allocator.free(c_include_path);
+            var it = mem.tokenize(u8, c_include_path, ":");
+            while (true) {
+                const dir = it.next() orelse break;
+                try self.addIncludeDir(dir);
+            }
+        } else |err| switch (err) {
+            error.InvalidUtf8 => {},
```

Comment:
This is `unreachable` since we're not on Windows,
```zig
error.InvalidUtf8 => unreachable, // only possible on Windows
```

However, if we want to handle the error anyway in case we're worried this code could be copy pasted somewhere else, I think `return e` would be better than ignoring this error which would cause the program to behave as if the environment variable isn't set.

Reply:
make it `unreachable` makes more sense to me

Reply:
I agree, `unreachable` is fine but requires the comment imo